File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/python/solana/policies/create_sol_message_policy.py
# Usage: uv run python solana/policies/create_sol_message_policy.py

import asyncio

from cdp import CdpClient
from dotenv import load_dotenv
from cdp.policies.types import (
    CreatePolicyOptions,
    SignSolMessageRule,
    SolMessageCriterion,
)

load_dotenv()

async def main():
    async with CdpClient() as cdp:
        # Create a policy that only allows signing messages starting with "CDP:"
        policy = CreatePolicyOptions(
            scope="account",
            description="Allow messages with CDP prefix only",
            rules=[
                SignSolMessageRule(
                    action="accept",
                    criteria=[
                        SolMessageCriterion(
                            type="solMessage",
                            match="^CDP:.*",
                        ),
                    ],
                ),
            ],
        )

        result = await cdp.policies.create_policy(policy=policy)

        print("Created sol message policy: ", result.id)


asyncio.run(main())

File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/typescript/solana/policies/createSolMessagePolicy.ts
// Usage: pnpm tsx solana/policies/createSolMessagePolicy.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();

// Create a policy that only allows signing messages starting with "CDP:"
const policy = await cdp.policies.createPolicy({
  policy: {
    scope: "account",
    description: "Allow messages with CDP prefix only",
    rules: [
      {
        action: "accept",
        operation: "signSolMessage",
        criteria: [
          {
            type: "solMessage",
            match: "^CDP:.*",
          },
        ],
      },
    ],
  },
});

console.log("Created sol message policy: ", policy.id);
